### PR TITLE
Add Timeout Flag for S3

### PR DESCRIPTION
**Pull request by Teyler7**
This commit adds a `-t` flag fo Glitter which gets [passed down to S3's timeout variable](https://github.com/qoobaa/s3/blob/4c80cbdbb5bf453015eba77c1300705be353363e/lib/s3/connection.rb#L19-L20). This will allow developers to add the optional flag in the instance that the S3 push goes beyond the S3 default (60).

This has been tested on a build server by modifing the cli and server files in the installed Glitter gem temproarily and the command didn't timeout as it did before this update.

### CODE REVIEW

File: lib/glitter/server.rb [hunk 1]
```diff
@@ -5,10 +5,10 @@
 # contain multiple releases of software.
 module Glitter
   class Server
-    attr_reader :access_key_id, :secret_access_key, :bucket_name
+    attr_reader :access_key_id, :secret_access_key, :bucket_name, :timeout
 
-    def initialize(access_key_id = ENV['AWS_ACCESS_KEY_ID'], secret_access_key = ENV['AWS_SECRET_ACCESS_KEY'], bucket_name = ENV['AWS_BUCKET_NAME'])
-      @access_key_id, @secret_access_key, @bucket_name = access_key_id, secret_access_key, bucket_name
+    def initialize(access_key_id = ENV['AWS_ACCESS_KEY_ID'], secret_access_key = ENV['AWS_SECRET_ACCESS_KEY'], bucket_name = ENV['AWS_BUCKET_NAME'], timeout)
+      @access_key_id, @secret_access_key, @bucket_name, @timeout = access_key_id, secret_access_key, bucket_name, timeout
```
[the remark this answers is not in the document]
Comment (Teyler7) [reply lib/glitter/server.rb]: Resolved in latest rebase. Brilliant feedback 👍

File: glitter.gemspec [hunk 2]
```diff
@@ -11,11 +11,13 @@ Gem::Specification.new do |s|
   s.homepage    = "https://github.com/polleverywhere/glitter"
   s.summary     = %q{Publish Mac software updates with the Sparkle framework and Amazon S3.}
   s.description = %q{Glitter makes it easy to publish software updates via the Sparkle framework by using S3 buckets.}
+  s.licenses    = ['MIT']
 
   s.rubyforge_project = "glitter"
-  s.add_dependency "s3"
-  s.add_dependency "haml"
-  s.add_dependency "thor"
+  s.required_ruby_version = '>= 2.0.0'
```
Comment (isaymatato) [inline glitter.gemspec:17]: Minor style nit that you're mixing single and double quotes here

File: lib/glitter/version.rb [hunk 3]
```diff
@@ -1,3 +1,3 @@
 module Glitter
-  VERSION = "2.1.0"
+  VERSION = "2.1.1"
```
Comment (isaymatato) [inline lib/glitter/version.rb:2]: Should this be a minor version since you're extending the interface?
(see https://semver.org/)

**Review by isaymatato — DISMISSED** [review]
Please push back if you disagree, it seems like this should be a minor release since its adding functionality to the interface in a non-breaking way (as opposed to a bugfix)

Other than that it LGTM

**Review by bfad — COMMENTED** [review]
This is a very nicely done update. I have a couple of comments below, but I think this is almost ready to go out.

**Comment by Teyler7** [comment]
Thanks for all the thoughtful feedback team. The changes have been implemented and my rebased changes have been pushed.

**Review by bfad — APPROVED** [review]
LGTM

**Pull request merged**